src/pages/ShowResource.tsx

```diff
@@ -18,46 +20,71 @@ function ShowResource({ location }: RouteComponentProps) {
 
   const { data: entity, isLoading, error, reload } = useAsync(
     getComitResourceFn,
-    { resourcePath, watch: location.pathname }
+    {
+      resourcePath,
+      watch: location.pathname
+    }
   );
+  const axiosError = error as AxiosError;
 
-  if (isLoading) {
-    return <CenteredProgress title="Fetching..." />;
-  } else if (
-    !error &&
-    entity &&
-    entity.class &&
-    entity.class.includes("swaps")
-  ) {
-    return (
+  const [showLoading, setShowLoading] = useState(false);
+  const [preventReload, setPreventReload] = useState(false);
```

Comment:
:thinking: should we negate this and track whether we are currently reloading and default to true?